src/coreclr/tools/aot/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -1869,7 +1878,19 @@ private static bool IsTypeSpecForTypicalInstantiation(TypeDesc t)
                     originalMethod = methodOnUnderlyingType;
                 }
 
-                MethodDesc directMethod = constrainedType.TryResolveConstraintMethodApprox(exactType, originalMethod, out forceUseRuntimeLookup);
+                MethodDesc directMethod;
+                if (isStaticVirtual)
+                {
+                    directMethod = constrainedType.ResolveVariantInterfaceMethodToStaticVirtualMethodOnType(originalMethod);
+                    if (directMethod == null)
+                    {
+                        throw new RequiresRuntimeJitException(originalMethod.ToString());
```

Comment:
Could you add a test to the version resiliency tests in src/tests/readytorun/tests so that if the static interface, and implementing type are not in the current R2R module, that if the implementation changes from a default implementation to an exact implementation that the generated R2R code referring to such a  thing remains correct?

Reply:
Added in the latest commit, please take a look to double-check if it matches your intent.